src/txn/txn_rollback_to_stable.c

```diff
@@ -1491,33 +1499,41 @@ __rollback_to_stable_btree_apply(WT_SESSION_IMPL *session)
             continue;
         }
 
-        /* Set this flag to return error instead of panic if file is corrupted. */
-        F_SET(session, WT_SESSION_QUIET_CORRUPT_FILE);
-        ret = __wt_session_get_dhandle(session, uri, NULL, NULL, 0);
-        F_CLR(session, WT_SESSION_QUIET_CORRUPT_FILE);
-
-        /*
-         * Ignore performing rollback to stable on files that does not exist or the files where
-         * corruption is detected.
-         */
-        if ((ret == ENOENT) ||
-          (ret == WT_ERROR && F_ISSET(S2C(session), WT_CONN_DATA_CORRUPTION))) {
-            __wt_verbose(session, WT_VERB_RECOVERY_RTS(session),
-              "ignore performing rollback to stable on %s because the file %s", uri,
-              ret == ENOENT ? "does not exist" : "is corrupted.");
-            continue;
-        }
-        WT_ERR(ret);
-
         /*
          * The rollback operation should be performed on this file based on the following:
-         * 1. The tree is modified.
+         * 1. The dhandle is present in the cache and tree is modified.
          * 2. The checkpoint durable start/stop timestamp is greater than the rollback timestamp.
          * 3. There is no durable timestamp in any checkpoint.
          * 4. The checkpoint newest txn is greater than snapshot min txn id
          */
-        if (S2BT(session)->modified || max_durable_ts > rollback_timestamp || prepared_updates ||
+        WT_WITH_HANDLE_LIST_READ_LOCK(session, (ret = __wt_conn_dhandle_find(session, uri, NULL)));
+
+        if (ret == 0 && S2BT(session)->modified)
+            perform_rts = true;
+
+        WT_ERR_NOTFOUND_OK(ret, false);
+
+        if (perform_rts || max_durable_ts > rollback_timestamp || prepared_updates ||
           !durable_ts_found || has_txn_updates_gt_than_ckpt_snap) {
+            /* Set this flag to return error instead of panic if file is corrupted. */
+            F_SET(session, WT_SESSION_QUIET_CORRUPT_FILE);
+            ret = __wt_session_get_dhandle(session, uri, NULL, NULL, 0);
```

Comment:
Maybe add a comment that even though we succeeded with ```__wt_conn_dhandle_find```, and hence session's dhandle points to the required dhandle. We need to obtain the dhandle through bringing it into the session cache.

The comment would help understand why we are calling __wt_session_get_dhandle, when we session->dhandle already points to the one needed.